data/list/basic.lean

```diff
@@ -1081,6 +1081,13 @@ theorem forall_mem_of_forall_mem_cons {p : α → Prop} {a : α} {l : list α}
   ∀ x ∈ l, p x :=
 (forall_mem_cons.1 h).2
 
+theorem forall_mem_singleton {p : α → Prop} {a : α} (h : p a) : ∀ x ∈ [a], p x :=
+λ x p, (mem_singleton.mp p).symm ▸ h
```

Comment:
can we have this as a biconditional? It should be provable by simp.

Reply:
Sure, I'll do that. I didn't do it before because I mainly needed `forall_mem_singleton` only for the construction of a singleton. Since `simp` already takes care of simplification without this theorem, I didn't see the need for a `↔` theorem.

Reply:
Done.